arduino/sketch/profiles.go

```diff
@@ -0,0 +1,213 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020-2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package sketch
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/arduino/arduino-cli/arduino/utils"
+	"github.com/arduino/go-paths-helper"
+	semver "go.bug.st/relaxed-semver"
+	"gopkg.in/yaml.v2"
+)
+
+// Project represents all the profiles defined for the sketch
+type Project struct {
+	Profiles       map[string]*Profile `yaml:"profiles"`
+	DefaultProfile string              `yaml:"default_profile"`
+}
+
+// AsYaml outputs the project file as Yaml
+func (p *Project) AsYaml() string {
+	res := "profiles:\n"
```

Comment:
I've got two issues:
- The ordering of the fields is not guaranteed
- Some elements are not the straight encoding of the underlying fields: for example, the `ProfileLibraryReference` is a struct with two fields (`library name` and `library version`) but it is encoded as a string `"library name (library version)"`. This can be easily implemented through a custom unmarshaler method `UnmarshalYAML` but the opposite seems quite difficult (or maybe... I didn't understand how to use it...).

Maybe I could give it another try, later...